### PR TITLE
Initial set of API Standards

### PR DESCRIPTION
**Pull request by dgmstuart**
These are largely based on the White House and Google standards, as well as on our conversation on Thursday 12th October

Please check that everything is basically sane and as we discussed. If significant additions need to be made then these can be done as further PRs

### CODE REVIEW

**Comment by dgmstuart** [comment]
Would be good to get a review from Andre as well but he hasn't been added to this github yet

File: README.md [hunk 1]
```diff
@@ -189,3 +189,92 @@ GET /v1/properties/:propertyReference/
   ]
 }
 ```
+
+## Create a new repair
+
+Create a repair request, with or without a list of work orders
+
+```
+POST /v1/properties/:id/repairs
```
Comment (dgmstuart) [inline README.md:213]: Note: This URL is an idea @leeky and I have been discussing. 

It makes sense on the basis that internal repairs are always for a property, and one can imagine `GET /v1/properties/:property_reference/repairs` returning all the repairs for a particular property at some point in the future. 

However, individual repairs don't need to be fetched in this scoped way: we only need to pass a repair reference:

```
GET /v1/repairs/:repair_request_reference
```
 and not the property reference as well - e.g.
```
GET /v1/properties/:property_reference/repairs/:repair_request_reference
```

I don't think we need/want to add that restriction.

**Review by LBHakairamkonda — COMMENTED** [review]
Shouldn't this be similar to getting the property details by property reference?
```json
 {
  "results": [{
    "requestReference": "08912445",
   "propertyReference": "00078345",
   "problemDescription": "The fan is buzzing and sometimes not spinning at all",
  "priority": "N",
  "contact": {
    "name": "Al Smith",
    "telephoneNumber": "07876543210",
    "emailAddress": "al.smith@hotmail.com",
    "callbackTime": "8am - 12pm"
  },
   "workOrders": [
     {
       "workOrderReference": "20090190",
       "sorCode":  "20090190",
       "supplierReference": "00000127"
     }
   ]
}
   ]
 }

**Comment by dgmstuart** [comment]
@LBHakairamkonda Which place did you mean? You can comment directly on the code by clicking the blue [+] icon when hovering over a line of code.

File: README.md [hunk 2]
```diff
@@ -2,3 +2,338 @@
...
+
+## Data Formats
+
+- Dates and Times should be formatted as RFC3339:
+  - Don’t include milliseconds
+  - Always use UTC (represented with Z)
+  - Where just a date is required, still provide a full time object, with the time as 00:00:00
+  - Example time: `"2017-11-06T16:34:41Z"`
+  - Example date: `"2017-11-06T00:00:00Z"`
+- Durations should be represented as a number of minutes:
```
Comment (leeky) [inline README.md:32]: [ISO8601](https://en.wikipedia.org/wiki/ISO_8601#Durations) also defines a standard for durations, not sure if we want to consider using it, or stick with minutes for simplicity's sake.

File: README.md [hunk 3]
```diff
@@ -2,3 +2,338 @@
...
+
+Many of the standards set out in this document are based on the
+[White House Web API Standards](https://github.com/WhiteHouse/api-standards),
+with some specifics from the
+[Google JSON StyleGuide](https://google.github.io/styleguide/jsoncstyleguide.xml#Date_Property_Values)
+
+## Data Properties
+
+- Property names should be in lower camel case
+  - Good: `"contactTelephoneNumber"`
```
Comment (leeky) [inline README.md]: Should these be left unquoted?

e.g. `contactTelephoneNumber` vs `"contactTelephoneNumber"`

File: README.md [hunk 4]
```diff
@@ -2,3 +2,338 @@
...
+- Property names should use full words, not abbreviations
+  - Good: `"propertyReference"`
+  - Bad: `"propertyRef”`
+
+## Data Formats
+
+- Dates and Times should be formatted as RFC3339:
+  - Don’t include milliseconds
+  - Always use UTC (represented with Z)
+  - Where just a date is required, still provide a full time object, with the time as 00:00:00
```
Comment (leeky) [inline README.md:29]: :+1:

**Review by leeky — COMMENTED** [review]
Just a couple of small comments, otherwise looks pretty good.

File: README.md (hunk 2 above)
Comment (dgmstuart) [reply README.md:32]: We discussed this and concluded that it looks like a standard which is comprehensive, but overengineered for our use case.

File: README.md (hunk 4 above)
Comment (dgmstuart) [reply README.md]: Yeah probably better.

File: README.md (hunk 2 above)
Comment (LBHAKovalichin) [reply README.md:32]: For hackney purposes any duration can be represented in minutes. The standard includes years and months and across services I do not see any duration that will exceed a few days. (Even if it does current convention is to use From and To dates for those)

File: README.md (hunk 4 above)
Comment (dgmstuart) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (dgmstuart) [reply README.md]: Yup - probably better: I've added a new commit

File: README.md (hunk 1 above)
Comment (LBHAKovalichin) [reply README.md:213]: This makes sense. If you already have a reference of a specific object you should not need to specify another.

**Comment by dgmstuart** [comment]
@LBHakairamkonda If you could clarify your comment, I can address and get this merged.

**Review by LBHakairamkonda — COMMENTED** [review]
Looks good to me

**Review by LBHakairamkonda — COMMENTED** [review]
Looks good to me

**Pull request merged**